lib/insights/api/common/open_api/generator.rb

```diff
@@ -198,9 +199,16 @@ def openapi_schema(klass_name)
 
           def openapi_list_description(klass_name, primary_collection)
             sub_collection = (primary_collection != klass_name)
+            klass = klass_name.constantize
+            operationId = if klass.respond_to?(:list_operation_id)
+                            klass.send(:list_operation_id)
+                          else
+                            "list#{primary_collection if sub_collection}#{klass_name.pluralize}"
+                          end
+            validate_operation_id(operationId, klass_name)
```

Comment:
I would like to see this block of code move into an `operation_id` method that takes parameters (`klass_name, primary_collection, sub_collection`) and can just be called from line 211 below.